### ISSUE TITLE
prepare for a CRAN release

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1912]
A change to Rcpp necessitates an oce update on CRAN.  The change for the Rcpp is at https://github.com/dankelley/oce/pull/1822/files, and of course there are also new oce features and bug fixes that have occurred since then.

I'll make a checklist, to keep track of my steps in the process.
* [ ] `spell_check()`
* [ ] `devtools::check(args = c('--run-donttest','--as-cran'))`
* [ ] `devtools::check_win_release()`
* [ ] `devtools::check_win_devel()`
* [ ] `devtools::check_win_oldrelease()`
* [ ] rhub checks
* [ ] revdepcheck
* [ ] `#devtools::install_github("r-lib/revdepcheck")`
* [ ] `revdepcheck::revdep_check(timeout=30*60,num_workers=4)`
* [ ] update `cran_comments.md`

**Comment by dankelley** [comment]
I'm not seeing why the revdep_check is failing, so I made an issue on its gh site (https://github.com/r-lib/revdepcheck/issues/322)

**Comment by dankelley** [comment]
I got revdep working.

**Comment by dankelley** [comment]
My plan is to release at 1300h today.  All the tests (in the first comment of this page) work, and they are quite extensive. 

**IMPORTANT NOTE** to any co-developer: it's the "master" branch that matters.  Once I submit, I'll tag that, and then merge it into "develop".  **If you're trying to build/test, please be sure to work on "master".**

**Comment by richardsc** [comment]
(separate issue, but perhaps it's time to go through the "master" to "main" branch name switch?)

https://github.com/github/renaming

**Comment by dankelley** [comment]
I thought about that long ago, but there were some problems (the details of which I forget).  However, things may be smoothed-out now.  That link you sent makes me think it would be okay to rename it, at least for people who get oce through GH.

I vote +0.75 on my trying to rename it.  I can't imagine I'd wreck anything.  Unless you vote a high negative number, I'll start to rename it at 1130.  That would give me time to also fix up the actions for continuous tests, and so forth.

**Comment by dankelley** [comment]
I used the web interface to rename it.  Then, to be cautious, I copied the whole directory on my filesystem and did "gh repo clone dankelley/oce" to start afresh.  

For some reason, now there are some diffs between "main" and "develop" that were not there in comparing "master" and "develop" before.  I'll just take this new "master" as the central concern, and rebuild it.  (I think what happened is that some Rd files got rebuilt, for some reason.)

Basically, I am going to redo the checklist in the first comment of this issue.  (I'll skip to the local build, of course, to get updated Rd files.)

**Comment by dankelley** [comment]
I found references to "master" in README.md and in .github/worflows/pkgdown.html, and so changed them to "main".  Using 'git grep', I didn't uncover any other things that looked problematic.

**Comment by dankelley** [comment]
All tests are complete.  Proceeding to `devtools::release()`.

**Comment by dankelley** [comment]
Submitted via `devtools::release()`.  This created `CRAN-SUBMISSION` at the top level.

**Comment by dankelley** [comment]
I got the email to confirm submission (well, my Junk folder got it, but I knew to check there).

**Comment by dankelley** [comment]
I got the email stating that it was submitted.  Now I think we wait for O(1) day until it gets examined by the first layer of QC machines.

**Comment by dankelley** [comment]
Things are going quickly today.  I just got the "thanks, package oce_1.5-0.tar.gz is on its way to CRAN" email.

**Comment by dankelley** [comment]
I see the source now at https://cran.r-project.org/web/packages/oce/index.html (at least, the mirror I receive when I click that link).  I'll check tomorrow to see if any binaries have become available.  Once I see all of them filled in, I'll close this issue.  That usually takes a week or so.  But, who knows, something odd might happen when the new Rcpp gets into CRAN.

**Comment by dankelley** [comment]
I'm seeing a warning for the macos build (https://www.r-project.org/nosvn/R.check/r-release-macos-x86_64/oce-00check.html) which has two components:

1.  a message about 1.5.0 being submitted although 1.5.0 is active.  I think this is a false positive of some type, because I did not submit it twice.
2. a message about a non-accessible URL (https://www.soest.hawaii.edu/gmt/gmt/html/GMT_Docs.html#x1-820004.15
).  I think this must be a temporary problem since I had done a URL-check just before submitting (literally, an hour before submitting).  I guess that URL might have failed permanently in that hour, but it seems more likely that it was temporarily unavailable.  I just checked the URL, and it worked.

I don't see the point of getting in CRAN "bad books" by trying a resubmission for a URL that was likely only temporarily unavailable, and certainly not for this false positive.  Let's see if other platforms give errors, before confusing things.
